sotodlib/toast/workflows/proc_mapmaker_filterbin.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2023-2023 Simons Observatory.
+# Full license can be found in the top level "LICENSE" file.
+"""Simultaneous filtering and binned mapmaking.
+"""
+
+import numpy as np
+from astropy import units as u
+import toast
+import toast.ops
+from toast.observation import default_values as defaults
+
+from .. import ops as so_ops
+
+
+def setup_mapmaker_filterbin(operators):
+    """Add commandline args, operators, and templates for TOAST FilterBin mapmaker.
+
+    Args:
+        operators (list):  The list of operators to extend.
+
+    Returns:
+        None
+
+    """
+    operators.append(toast.ops.FilterBin(name="filterbin", enabled=False))
+
+
+def mapmaker_filterbin(job, otherargs, runargs, data):
+    """Run the TOAST FilterBin mapmaker.
+
+    Args:
+        job (namespace):  The configured operators and templates for this job.
+        otherargs (namespace):  Other commandline arguments.
+        runargs (namespace):  Job related runtime parameters.
+        data (Data):  The data container.
+
+    Returns:
+        None
+
+    """
+    log = toast.utils.Logger.get()
+    timer = toast.timing.Timer()
+    timer.start()
+
+    # Configured operators for this job
+    job_ops = job.operators
+
+    job_ops.filterbin.binning = job_ops.binner_final
+    job_ops.filterbin.det_data = job_ops.sim_noise.det_data
+    job_ops.filterbin.output_dir = otherargs.out_dir
+
+    if job_ops.filterbin.enabled:
```

Comment:
The reason I did this was so that the operator was configured with information from other operators, even if the actual run was disabled.  In principle one could then re-enable and run it, although agreed that would likely not be done very often.

Reply:
I ended up implementing exactly what you suggest- the important point that I missed is the case where multiple workflow functions are coupled together and expect their traits to be set consistently.  So it is safer to not set traits based on other operators unless we need to.